packages/babel-parser/src/parser/expression.js

```diff
@@ -594,163 +602,205 @@ export default class ExpressionParser extends LValParser {
     state: N.ParseSubscriptState,
```

Comment:
`parseSubscript` has been broken into smaller pieces. I find it more easier to go through after refactoring: https://github.com/JLHwung/babel/blob/ee4d63ef00e5090f44ffedee718dd18fdc6e21e0/packages/babel-parser/src/parser/expression.js#L597